### DISCUSSION TITLE
Decoupling frontend & backend

### DISCUSSION THREAD

**Discussion by adarshpunj** [zauberzeug/nicegui#3420] [Ideas / Feature Requests]
Hi, 

I've been exploring ways to decouple frontend and backend in a NiceGUI project. If not on a repo level, then at least on a directory level. 

I'm not sure if it's a good design but sometimes I'd want to build a vanilla FastAPI backend, then later on find a frontend for it (NiceGUI in our case). There will be no sessions and only JWT token-based auth.

Did we intend NiceGUI to be used in this way? Is there a sample repo/project that does this?

**Comment by python-and-novella** [comment]
https://github.com/zauberzeug/nicegui/tree/main/examples/fastapi

**Comment by Rezanmonis** [comment]
Hello

We are looking for frontend and backend developers as our teammates for our edtech project rezanmonis@gmail.com
